test/libdevcore/rlp.cpp

```diff
@@ -40,8 +40,119 @@ namespace js = json_spirit;
 namespace dev
 {
 	namespace test
-	{
-		static void buildRLP(js::mValue& _v, RLPStream& _rlp)
+	{	
+		void buildRLP(js::mValue& _v, RLPStream& _rlp);
+		void checkRLPAgainstJson(js::mValue& v, RLP& u);
+
+		void doRlpTests(json_spirit::mValue& v, bool _fillin)
+		{
+			for (auto& i: v.get_obj())
+			{
+				js::mObject& o = i.second.get_obj();
+				if (test::Options::get().singleTest && test::Options::get().singleTestName != i.first)
+				{
+					o.clear();
+					continue;
+				}
+
+				std::cout << "  " << i.first << std::endl;
+				TBOOST_REQUIRE((o.count("out") > 0));
+				TBOOST_REQUIRE((!o["out"].is_null()));
+
+				if (_fillin)
+				{
+					try
+					{
+						bytes payloadToDecode = fromHex(o["out"].get_str());
+						RLP payload(payloadToDecode);
+						if (payload.isEmpty())
+							BOOST_THROW_EXCEPTION(RLPException() << errinfo_comment("Decoded Empty RLP!"));
+						o["in"] = "VALID";
+					}
+					catch (Exception const& _e)
+					{
+						cnote << "Exception: " << diagnostic_information(_e);
+						o["in"] = "INVALID";
+					}
+					catch (std::exception const& _e)
+					{
+						cnote << "rlp exception: " << _e.what();
+						o["in"] = "INVALID";
+					}
+				}
+				else
+				{
+					//Check Encode
+					TBOOST_REQUIRE((o.count("in") > 0));
+					int skipEncode = 0;
```

Comment:
because code skiped in two diffrent scenarious. 
1) if invalid test
2) if valid test
the logic of expetions is diffrent below